operator/pkg/helmreconciler/common.go

```diff
@@ -125,5 +126,77 @@ func applyOverlay(current, overlay runtime.Object) error {
 	if err != nil {
 		return err
 	}
-	return runtime.DecodeInto(unstructured.UnstructuredJSONScheme, merged, current)
+	// save any immutable values
+	clusterIP := getPath(current, name.ServiceStr, util.PathFromString("spec.clusterIP"))
```

Comment:
I don't know what all of the immutable values are but this probably isn't the full list as you mentioned. not sure if there is a better option though...

Reply:
pretty sure it is not the full list. unfortunately i couldn't find any reference doc or field annotation that indicates immutability. it only shows up in the comments and in random places in docs.